connect/api/src/main/java/org/apache/kafka/connect/data/Values.java

```diff
@@ -704,7 +711,9 @@ protected static void appendIterable(StringBuilder sb, Iterator<?> iter) {
     }
 
     protected static String escape(String value) {
-        return value.replaceAll("\\\\", "\\\\\\\\").replaceAll("\"", "\\\\\"");
+        return REPLACE_DOUBLEQOUTE.matcher(
+                REPLACE_BACKSLASH_TWO.matcher(value).replaceAll("\\\\\\\\")
```

Comment:
In OAuthBearerUnsecuredLoginCallbackHandler.java you use Matcher.quoteReplacement. Why do you do it differently here?

Reply:
Looks like an oversight. The first thing the source code of Matcher.quoteReplacement does is this:
```
if ((s.indexOf('\\') == -1) && (s.indexOf('$') == -1))
            return s;
```
So anything with a `\` gets processed and therefor needs it.

That being said, one should not rely on knowledge of source code to assume behavior. I'll add them everywhere to be consistent.

Reply:
Scratch that, now that I'm looking at the actual code, it because of the difference between `replace` and `replaceAll` in String.java.

`replace` will do `Matcher.quoteReplacelemt`, `replaceAll` doesn't. And the javadoc for `replaceAll` warns the user of this and suggests using it beforehand if desired.